### PR TITLE
Fix bug with blankslate having 0 max depth recursion

### PR DESCRIPTION
**Pull request by ldemailly**
(also adding comment missing from #139 reviews)

### CODE REVIEW

File: eval/eval_api.go [hunk 1]
```diff
@@ -125,12 +127,15 @@ func EvalString(this any, code string, emptyEnv bool) (object.Object, error) {
 	if len(p.Errors()) != 0 {
 		return object.NULL, fmt.Errorf("parsing error: %v", p.Errors())
 	}
-	var evalState *State
+	evalState, ok := this.(*State)
```
Comment (ccoVeille) [inline eval/eval_api.go:130]: I'm only noticing it now, but a variable named this/self looks odd in Go code.

Comment (ldemailly) [reply eval/eval_api.go:130]: why ? it is "this" in this case (ie the eval state passed back through an any arg)

Comment (ccoVeille) [reply eval/eval_api.go:130]: this like self looks like a reference to current object, as in PHP or JavaScript

Comment (ccoVeille) [reply eval/eval_api.go:130]: But I might be the only one having nightmare about these languages

Comment (ldemailly) [reply eval/eval_api.go:130]: yet.. that's what it is here (*State) is the receiver in the rest of the package thus the name

**Pull request merged**